Feat/#10 ci/cd 적용

heyoni opened the pull request:
CI/CD 적용 및 NCP 서버 이전
- #10 
- #11 
- #12
- #19

heyoni committed:
feat: #11

- 테스트코드 자동 동작(git action) 추가

heyoni committed:
feat: #12 docker 도입

heyoni committed:
Merge branch 'feat/#10-CI/CD-적용' of https://github.com/f-lab-edu/What-cat-eat into feat/#10-CI/CD-적용

heyoni committed:
fix: git action 수정

heyoni committed:
fix: 테스트코드 오류로 인한 코드 수정

heyoni committed:
feat: docker 실행용 파일 추가, 필요 없는 파일 삭제

heyoni committed:
fix: github action 파일 추가

heyoni committed:
github action 테스트

heyoni committed:
github action 테스트

heyoni committed:
github action 테스트

heyoni committed:
github action 테스트

heyoni committed:
fix: github action 설정 수정, dockerfile 수정

- github action 동작 안되는 문제 해결(branches -> branches-ignore로 변경)
- dockerfile 중 폴더 위치 변경, 포트번호 변경

mihosang commented on dockerfile:
```diff
@@ -0,0 +1,11 @@
+
+COPY ./ /What-cat-eat
+
+CMD ["uvicorn", "main:app", "--host", "0.0.0.0", "--port", "80"]
```
docker build  후 docker run 해보셨을까요?
DB 연결에서 오류가 납니다
DB 연결에 필요한 정보를 어떻게 넘겨야 해결할 수 있을지 조사해보세요

mihosang commented on repositories/UserRepository.py:
```diff
@@ -65,7 +65,7 @@ def create(self, user_new: User):
         return self._user[user_new.id - 1]
 
-    def get(self, id):
+    def get(self, user, id):
         user = self._user[id - 1]
```
parameter 의 user, local 변수로서의 user가 중복

f-lab-zion replied:
param 의 user와 local 변수의 user가 중복

mihosang reviewed (changes requested):
.

f-lab-zion commented on dockerfile:
```diff
@@ -0,0 +1,11 @@
+
+COPY ./ /What-cat-eat
+
+CMD ["uvicorn", "main:app", "--host", "0.0.0.0", "--port", "80"]
```
docker build & run  해보셨을까요?
DB 연결에 필요한 정보를 전달해줘야 하는데 어떤 방법을 사용할 수 있는지 조사해보세요

heyoni replied:
넵 찾아보겠습니다!

f-lab-zion replied:
혹시 이것도 해결하셨나요?

heyoni replied:
아직 찾아보고 있는 중인데 작성 후 이번 PR에 같이 올릴까요?

f-lab-zion reviewed (changes requested):
몇가지 확인 부탁드립니다

heyoni committed:
Merge branch 'develop' into feat/#10-CI/CD-적용

heyoni committed:
feat: 서버 이전, 코드리뷰 반영

- AWS -> NCP 서버이전
- docker 실행 방법 추가
- pre-commit-config.yaml 파일 오류 수정
- Repository get 요청 시 params 중복 수정

heyoni committed:
fix: black 에러 수정, yaml 파일 수정

heyoni committed:
fix: 잘못 올라간 파일 삭제, github action 수정

heyoni committed:
fix:github action 수정

heyoni committed:
fix:github action 수정

heyoni committed:
fix:github action 수정

heyoni committed:
fix:github action 수정

heyoni committed:
fix:github action 수정

heyoni committed:
fix:github action 수정

heyoni committed:
fix:github action 수정

heyoni committed:
fix:github action 수정

heyoni committed:
fix:github action 수정

heyoni committed:
fix: github action 수정

heyoni committed:
fix: github action 수정

f-lab-zion reviewed (approved):
👍 수고하셨습니다 Docker 파일은 다른 PR에서 따로 확인하겠습니다
